lib/project_types/script/layers/infrastructure/api_clients/partners_proxy_api_client.rb

```diff
@@ -0,0 +1,55 @@
+module Script
+  module Layers
+    module Infrastructure
+      module ApiClients
+        class PartnersProxyApiClient
+          def initialize(ctx, api_key)
+            @ctx = ctx
+            @api_key = api_key
+          end
+
+          def query(query_name, variables: {})
+            proxy_response = ShimAPI.query(@ctx, query_name, api_key: @api_key, variables: variables.to_json)
+            raise_if_graphql_failed(proxy_response)
+
+            response = proxy_response["data"]["scriptServiceProxy"]
+            JSON.parse(response) if response
```

Comment:
That would make sense, would likely create a safe guard in the event of partners/script service having a bug.

Reply:
I think it would make sense to do this in a separate PR to get this one out. The error needs to be surfaced with a message, and I'm not sure telling the user that the response was invalid is the right thing to do as this would likely be caused by something being wrong on our end.